substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/GCImpl.java

```diff
@@ -207,6 +209,9 @@ private boolean collectImpl(GCCause cause, long requestingNanoTime, boolean forc
         precondition();
 
         NoAllocationVerifier nav = noAllocationVerifier.open();
+
+        JfrGCPausePhase phase = StackValue.get(JfrGCPausePhase.class);
```

Comment:
I think that a large portion of this PR is unnecessary. I don't think that there is any need for extra data structures:
- the GC epoch can be queried at any time via getCollectionEpoch()
- the start ticks can be stored in a local
- the level can be a constant
- the phase name can be a String constant

Reply:
Question: Are constant strings guaranteed in unmovable partition of heap in binary image? I introduced NativeString, not because of this PR, but if I want to generate per-worker events during, e.g. object evacuation phase, I am not sure, but unlikely, there are barriers for accessing the phase names on GC path.

Reply:
Constant strings are located the image heap and won't be touched by the GC, so they can be used safely. Per-worker events would still use the same constant Strings (only the data in the `eventThread` field of the JFR events would be different).